tests/integration/targets/ios_vrf_interfaces/tests/common/gathered.yaml

```diff
@@ -0,0 +1,38 @@
+---
+- ansible.builtin.debug:
+    msg: START ios_vrf_interfaces gathered integration tests on connection={{ ansible_connection }}
+
+- ansible.builtin.include_tasks: _populate_config.yaml
+
+- block:
+    - name: Gathered the provided configuration with the existing running configuration
+      register: result
+      cisco.ios.ios_vrf_interfaces:
+        config:
+        state: gathered
+
+    - name: Debug gathered.config
+      ansible.builtin.debug:
+        var: gathered.config
+
+    - name: Debug result.gathered
+      ansible.builtin.debug:
+        var: result.gathered
+
+    - name: Run show running-config | section ^interface
+      cisco.ios.ios_command:
+        commands: show running-config | section ^interface
+      register: config
+
+    - name: Assert
+      ansible.builtin.assert:
+        that:
+          - gathered.config | length == result.gathered | length
```

Comment:
this is a complicated way to compare it, can be simple

Reply:
aah i see this test i have written for my understanding , Now i have updated with the test with simplified logic